plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/files/FileSlice.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hudi.files;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.TreeSet;
+
+import static java.util.Objects.requireNonNull;
+
+public class FileSlice
+        implements Serializable
+{
+    private final String baseInstantTime;
+
+    private Optional<HoodieBaseFile> baseFile;
+
+    private final TreeSet<HoodieLogFile> logFiles;
+
+    public FileSlice(String baseInstantTime)
+    {
+        this.baseInstantTime = requireNonNull(baseInstantTime, "baseInstantTime is null");
+        this.baseFile = Optional.empty();
+        this.logFiles = new TreeSet<>(HoodieLogFile.getReverseLogFileComparator());
+    }
+
+    public void setBaseFile(HoodieBaseFile baseFile)
+    {
+        this.baseFile = Optional.ofNullable(baseFile);
+    }
+
+    public void addLogFile(HoodieLogFile logFile)
+    {
+        this.logFiles.add(logFile);
+    }
+
+    public String getBaseInstantTime()
+    {
+        return baseInstantTime;
+    }
+
+    public Optional<HoodieBaseFile> getBaseFile()
+    {
+        return baseFile;
+    }
+
+    public boolean isEmpty()
+    {
+        return (baseFile == null) && (logFiles.isEmpty());
+    }
+
+    @Override
+    public String toString()
+    {
+        return "FileSlice {" +
+            "baseCommitTime=" + baseInstantTime +
+            ", baseFile='" + baseFile + '\'' +
+            ", logFiles='" + logFiles + '\'' +
+            '}';
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        FileSlice slice = (FileSlice) o;
+        return Objects.equals(baseInstantTime, slice.baseInstantTime)
+                && Objects.equals(baseFile, slice.baseFile) && Objects.equals(logFiles, slice.logFiles);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(baseInstantTime);
```

Comment:
This doesn't match equals. Can we remove?

Reply:
It was copied from hudi library, it is the same case as in CompactionOperation, I will try to remove both and see if tests pass

Reply:
removed